straxen/plugins/event_processing.py

```diff
@@ -560,45 +583,54 @@ def setup(self):
-            r_cor = r_obs + delta_r
-            scale = r_cor / r_obs
-
-        # z correction due to longer drift time for distortion
-        # (geometrical reasoning not valid if |delta_r| > |z_obs|,
-        #  as cathetus cannot be longer than hypothenuse)
-        with np.errstate(invalid='ignore'):
-            z_cor = -(z_obs ** 2 - delta_r ** 2) ** 0.5
-            invalid = np.abs(z_obs) < np.abs(delta_r)
-            # do not apply z correction above gate
-            invalid |= z_obs >= 0
-        z_cor[invalid] = z_obs[invalid]
-        delta_z = z_cor - z_obs
-
-        result.update({'x': orig_pos[:, 0] * scale,
-                       'y': orig_pos[:, 1] * scale,
-                       'r': r_cor,
-                       'r_naive': r_obs,
-                       'r_field_distortion_correction': delta_r,
-                       'theta': np.arctan2(orig_pos[:, 1], orig_pos[:, 0]),
-                       'z_naive': z_obs,
-                       # using z_obs in agreement with the dtype description
-                       # the FDC for z (z_cor) is found to be not reliable (see #527)
-                       'z': z_obs,
-                       'z_field_distortion_correction': delta_z
-                       })
+       
+	# s_i == 0 indicates the main event, while s_i != 0 means alternative S1 or S2 is used based on s_i value 
+        for s_i in [0, 1, 2]:
+            
+            drift_time = events['drift_time']  if not s_i else events[f'alt_s{s_i}_interaction_drift_time']
+            
+            z_obs = - self.electron_drift_velocity * drift_time
+            xy_pos = 's2_' if s_i != 2 else 'alt_s2_'
+            orig_pos = np.vstack([events[f'{xy_pos}x'], events[f'{xy_pos}y'], z_obs]).T
+            r_obs = np.linalg.norm(orig_pos[:, :2], axis=1)
+            delta_r = self.map(orig_pos)
+            z_obs = z_obs + self.electron_drift_velocity * self.electron_drift_time_gate
+
+            # apply radial correction
+            with np.errstate(invalid='ignore', divide='ignore'):
+                r_cor = r_obs + delta_r
+                scale = r_cor / r_obs
+
+            # z correction due to longer drift time for distortion
+            # (geometrical reasoning not valid if |delta_r| > |z_obs|,
+            #  as cathetus cannot be longer than hypothenuse)
+            with np.errstate(invalid='ignore'):
+                z_cor = -(z_obs ** 2 - delta_r ** 2) ** 0.5
+                invalid = np.abs(z_obs) < np.abs(delta_r)
+                # do not apply z correction above gate
+                invalid |= z_obs >= 0
+            z_cor[invalid] = z_obs[invalid]
+            delta_z = z_cor - z_obs
+            
+            pre_field = '' if s_i == 0  else f'alt_s{s_i}_'
+            post_field = '' if s_i == 0  else '_fdc'
+            result.update({f'{pre_field}x{post_field}': orig_pos[:, 0] * scale,
+                           f'{pre_field}y{post_field}': orig_pos[:, 1] * scale,
```

Comment:
I think you have an edge case here. In case x/y is 0 and dr is zero too. You will get here x/y = np.nan. Because youe scale will be 0/0. Might be unlikely but I think it is better to avoid this edge case.

Reply:
True, on the other hand this is true for numerical zeros, very unlike for floating points. I can take care of it forcing it to zero, but I do really think this edge case might actually never happen

Reply:
Yes, but I would prefer to fix it. These things usually tend to appear at some point. And it is not difficult to fix.